src/FoundationClasses/TKernel/OSD/OSD_PerfMeter.hxx

```diff
@@ -16,65 +16,61 @@
-  OSD_PerfMeter()
-      : myIMeter(-1)
-  {
-  }
+  //! Constructs a void meter (to further call Init and Start).
+  Standard_EXPORT OSD_PerfMeter() = default;
 
-  //! Constructs and starts (if autoStart is true) the named meter
-  OSD_PerfMeter(const char* theMeter, const bool theToAutoStart = true)
-      : myIMeter(perf_get_meter(theMeter, 0, 0))
-  {
-    if (myIMeter < 0)
-      myIMeter = perf_init_meter(theMeter);
-    if (theToAutoStart)
-      Start();
-  }
+  //! Constructs and starts (if autoStart is true) the named meter.
+  //! @param theMeterName Name of the meter. If the meter with such name was already created,
+  //!        an haven't been killed, it will be used.
+  //! @param theToAutoStart If true, the meter will be started immediately after creation.
+  //!        Otherwise, the user should call Start() method to start the meter.
+  //!        Note that if meter already exists, theToAutoStart == true will reset it.
+  Standard_EXPORT OSD_PerfMeter(const std::string& theMeterName, const bool theToAutoStart = true);
 
-  //! Prepares the named meter
-  void Init(const char* theMeter)
-  {
-    myIMeter = perf_get_meter(theMeter, 0, 0);
-    if (myIMeter < 0)
-      myIMeter = perf_init_meter(theMeter);
-  }
+  //! Destructs this meter handler. Note that the meter itself is not destructed
+  //! and will be printed on stdout upon finish of the program. The meter with the name
+  //! Specified in the constructor can still be used in other places of the code.
+  Standard_EXPORT ~OSD_PerfMeter();
 
-  //! Starts the meter
-  void Start() const { perf_start_imeter(myIMeter); }
+  //! Prepares the named meter. If the meter with such name was already created,
+  //! it will be used. Otherwise, a new meter will be created.
+  Standard_EXPORT void Init(const std::string& theMeterName);
 
-  //! Stops the meter
-  void Stop() const { perf_stop_imeter(myIMeter); }
+  //! Starts the meter. If the meter was already started, it will be reset.
+  //! Note that the meter with the name specified in the constructor can still be used
+  //! in other places of the code.
+  Standard_EXPORT void Start() const;
 
-  //! Increments the counter w/o time measurement
-  void Tick() const { perf_tick_imeter(myIMeter); }
+  //! Stops the meter.
+  Standard_EXPORT void Stop() const;
 
-  //! Outputs the meter data and resets it to initial state
-  void Flush() const { perf_close_imeter(myIMeter); }
+  //! Returns the elapsed time in milliseconds since the meter was started.
+  Standard_EXPORT double Elapsed() const;
 
-  //! Assures stopping upon destruction
-  ~OSD_PerfMeter()
-  {
-    if (myIMeter >= 0)
-      Stop();
-  }
+  //! Outputs the meter data and resets it to initial state.
+  Standard_EXPORT void Kill() const;
 
-private:
-  OSD_PerfMeter(const OSD_PerfMeter&);
-  OSD_PerfMeter& operator=(const OSD_PerfMeter&);
+  //! Prints the data of this meter.
+  Standard_EXPORT std::string Print() const;
```

Comment:
std::string is not recommended to be out of header. Only internal usage is accepted.